interface/repository/task_repository.go

```diff
@@ -16,12 +18,19 @@ type TaskRepository interface {
 }
 
 // NewTaskRepository creates a new TaskRepository with the given Database
-func NewTaskRepository(database datastore.DB) TaskRepository {
-	return &taskRepository{database: database}
+func NewTaskRepository(database datastore.DB, client client.ApiClient) TaskRepository {
+	return &taskRepository{database: database, client: client}
 }
 
 func (tr *taskRepository) FindAll(t []*model.Task) ([]*model.Task, error) {
-	return tr.database.FindAll(), nil
+	t, err := tr.client.RequestAllTasks(t)
+	if err != nil {
+		return nil, err
+	}
+
+	go tr.database.SaveRecords(t)
```

Comment:
It's not a bad idea to use a goroutine to save the records without blocking the main flow, my only concern would be: how are you handling the errors that could result from this function?

Reply:
Actually I do not have a clear idea of how to handle the errors in this case, an option would be handle error inside the SaveRecords method and log the error. Do you have any suggestion to handle errors in this kind of situation? 🤔

Reply:
Yes, that's right. Another option would be to just do everything synchronously. 👍🏽